src/util.cpp

```diff
@@ -726,19 +726,14 @@ void RenameThread(const char* name)
 
 void SetupEnvironment()
 {
-    std::locale loc("C");
-    // On most POSIX systems (e.g. Linux, but not BSD) the environment's locale
-    // may be invalid, in which case the "C" locale is used as fallback.
-#if !defined(WIN32) && !defined(MAC_OSX) && !defined(__FreeBSD__) && !defined(__OpenBSD__)
-    try {
-        loc = std::locale(""); // Raises a runtime error if current locale is invalid
-    } catch (const std::runtime_error&) {
+    if (!setlocale(LC_ALL, ""))
+    {
+#if !defined(WIN32)
         setenv("LC_ALL", "C", 1);
-    }
 #endif
-    // The path locale is lazy initialized and to avoid deinitialization errors 
```

Comment:
Please correct instead of remove the comments. Esoteric knowledge about locales is needed to understand this on reading, heck, *we* will have forgotten about the rationale for this in a few months.